File: patientManagementSystem/src/main/java/com/example/patientManagementSystem/patientDetails/model/Patient.java
package com.example.patientManagementSystem.patientDetails.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;


@Document
@AllArgsConstructor
@NoArgsConstructor
@Data
public class Patient {

    //Basic entity of the patient details with id as primary key, name, gender, address, contact number, email
    @Id
    private String id;
    private String name;
    private Gender gender;
    private String contactNumber;
    private String email;
}

File: patientManagementSystem/webapp/src/app/model/patient.ts
export class Patient {
    id?: string;
    name: string;
    gender: any;
    age: number;
    bloodGroup: string;
    contactNumber: string;
    email: string;
    address: string;
    description: string;
  
    constructor(
      name: string,
      gender: any,
      age: number,
      bloodGroup: string,
      contactNumber: string,
      email: string,
      address: string,
      description: string
    ) {
      this.name = name;
      this.gender = gender;
      this.age = age;
      this.bloodGroup = bloodGroup;
      this.contactNumber = contactNumber;
      this.email = email;
      this.address = address;
      this.description = description;
    }
  }
  